contracts/claimdrops.js

```diff
@@ -0,0 +1,346 @@
+      params.maxExpiryTime = maxExpiryTime;
+    }
+    if (maxExpiresPerBlock) {
+      if (!api.assert(Number.isInteger(maxExpiresPerBlock) && maxExpiresPerBlock > 0 && maxExpiresPerBlock <= 1000, 'invalid maxExpiresPerBlock')) return;
+      params.maxExpiresPerBlock = maxExpiresPerBlock;
+    }
+
+    await api.db.update('params', params);
+  }
+};
+
+const hasValidPrecision = (value, precision) => (api.BigNumber(value).dp() <= precision);
+
+const transferIsSuccessful = (result, action, from, to, symbol, quantity) => {
+  if (result.errors === undefined
+    && result.events && result.events.find(el => el.contract === 'tokens'
+    && el.event === action
+    && el.data.from === from
+    && el.data.to === to
+    && api.BigNumber(el.data.quantity).eq(quantity)
+    && el.data.symbol === symbol) !== undefined) {
+    return true;
+  }
+
+  return false;
+};
+
+const getTimestamp = (value) => {
+  try {
+    const date = new Date(`${value}.00Z`);
+    return date.getTime();
+  } catch (e) {
+    return false;
+  }
+};
+
+const validateList = (list, precision) => {
+  const parsedList = [];
+  for (let i = 0; i < list.length; i += 1) {
+    const { 0: account, 1: limit } = list[i];
+
+    // account & limit validation
+    if (api.assert(account, `list[${i}]: account name cannot be undefined`)
+      && api.assert(api.isValidAccountName(account), `list[${i}]: invalid account name`)
+      && api.assert(limit, `list[${i}]: limit cannot be undefined`)
+      && api.assert(!api.BigNumber(limit).isNaN(), `list[${i}]: invalid limit`)
+      && api.assert(api.BigNumber(limit).gt(0), `list[${i}]: limit must be positive`)
+      && api.assert(hasValidPrecision(limit, precision), `list[${i}]: limit precision mismatch`)) {
+      parsedList.push({
+        account,
+        limit,
+      });
+    } else break;
+  }
+
+  // list validation, check if all values from list are valid & pushed into parsedList
+  if (api.assert(list.length > 0, 'list cannot be empty') && parsedList.length === list.length) return parsedList;
+  return false;
+};
+
+const isValidowner = async (name, type) => {
+  if (type === 'user') {
+    if (api.isValidAccountName(name)) return true;
+  } else if (type === 'contract') {
+    const contract = await api.db.findContract(name);
+    if (contract) return true;
+  }
+
+  return false;
+};
+
+actions.create = async (payload) => {
+  const {
+    symbol,
+    price,
+    pool,
+    maxClaims,
+    expiry,
+    owner,
+    ownerType,
+    list,
+    limit,
+    isSignedWithActiveKey,
+  } = payload;
+
+  if (api.assert(isSignedWithActiveKey === true, 'you must use a custom_json signed with your active key')
+    && api.assert(symbol && typeof symbol === 'string'
+      && price && typeof price === 'string' && !api.BigNumber(price).isNaN()
+      && pool && typeof pool === 'string' && !api.BigNumber(pool).isNaN()
+      && maxClaims && Number.isInteger(maxClaims)
```

Comment:
Can you do a max size check for maxClaims?  Should make sure it is less than Number.MAX_SAFE_INTEGER